chore: build production package by default

tltv opened the pull request:
Changes Maven configuration to build production package by default with 'mvn package' and run in development mode with 'jetty:run'.

PartOf: vaadin/flow#22715

tltv committed:
chore: build and run production mode by default

Changes Maven configuration to activate production profile by default and adds development profile that adds vaadin-dev.

PartOf: vaadin/flow#22715

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/vaadin/skeleton-starter-flow?pullRequest=953) <br/>All committers have signed the CLA.

tltv commented:
jetty drops optional dependency in same way as spring-boot does.. Which means that this setup can be actually simplified to exactly same as with spring-boot.

tltv committed:
chore: build production package by default

Changes Maven configuration to build production package by default with 'mvn package' and run in development mode with 'jetty:run'.

PartOf: vaadin/flow#22715

tltv committed:
update to 25.0.0-beta5

tltv committed:
bump to Vaadin 25.0.0-beta6

mstahv commented:
The changes themselves looks very good 👍 When testing with jetty plugin I get jna/copilot originated exceptions and the dev mod don't really work (at least the copilot icon doesn't appear).

mshabarov commented on pom.xml:
```diff
@@ -158,29 +156,22 @@
     <profiles>
         <profile>
-            <!-- Production mode is activated using -Pproduction -->
+            <!-- Production mode is activated by -Pproduction -->
```
Could we "embed" the vaadin plugin config from this profile to the "it" profile so to achieve one profile less?
